### PR TITLE
Mrtydi mdpr index

### PR DESCRIPTION
**Pull request by crystina-z**
1. the mdpr index on mrtydi v1.1 and corresponding readme; 
2. the script (to merge the faiss indexes generated per shard;
   - merge_faiss_index.py
3. add a `--delimiter` argument for the `encode` module, use to parse the fields under `doc['contents']`
   - pyserini/encode/_\_main__.py
   - pyserini/encode/_base.py
4. allow the directory extracted from tarball different from the specified index name, as long as there is only one directory under it. This is for the case when ppl didn't name the file correctly before compressing it into tarball and upload to server, which is costly to redo. **probably need more review attention**
   - pyserini/util.py

### CODE REVIEW

**Comment by lintool** [comment]
@MXueguang can you take a look first? I'll take a pass after.

**Review by MXueguang — APPROVED** [review]

File: pyserini/resources/index-metadata/faiss.mrtydi-v1.1-arabic.20220207.5df364.README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+corpus=mrtydi-v1.1-$lang/collection/docs.jsonl
+index_dir=mrtydi-mdpr-dindex/$lang
+
+wget https://git.uwaterloo.ca/jimmylin/mr.tydi/-/raw/master/data/$tarfn
+tar –xvf $tarfn
+gzip -cvf $corpus.gz > $corpus
+
+mkdir -p $index_dir
+
+python -m pyserini.encode input   --corpus $corpus \
```
Comment (lintool) [inline pyserini/resources/index-metadata/faiss.mrtydi-v1.1-arabic.20220207.5df364.README.md:18]: Minor nit - better alignment of the command options?

**Review by lintool — APPROVED** [review]
Minor nits - please address and then just go ahead and merge. Thanks @crystina-z !

**Comment by crystina-z** [comment]
all reviews fixed!

**Pull request merged**